src/ngraph/distributed.cpp

```diff
@@ -14,90 +14,35 @@
 // limitations under the License.
 //*****************************************************************************
 
-#ifdef NGRAPH_DISTRIBUTED_ENABLE
-
-#ifdef NGRAPH_DISTRIBUTED_MLSL_ENABLE
-#include <mlsl.hpp>
-#elif NGRAPH_DISTRIBUTED_OMPI_ENABLE
-#include <mpi.h>
-#endif
-
 #include "ngraph/distributed.hpp"
+#include "ngraph/distributed/mlsl.hpp"
+#include "ngraph/distributed/null.hpp"
+#include "ngraph/distributed/open_mpi.hpp"
```

Comment:
One problem is that we cannot build the code with mlsl and mpi simultaneously because mlsl and mpi will use the same mpirun command which will cause runtime error for linking to two libraries.

Reply:
Both headers are conditionalized on mlsl/openmpi, only one of which can be enabled in the build configuration.